scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,73 @@
+package com.twitter.scalding
+
+object ArgOptions extends Enumeration {
+  type ArgOptions = Value
+  val RequiredArg, OptionalArg, ListArg, BooleanArg = Value
+}
+
+class DescriptionArgs(override val m: Map[String, List[String]]) extends Args(m){
+  import ArgOptions._
+
+  private val map = new scala.collection.mutable.HashMap[String, (String, ArgOptions)]()
```

Comment:
Does this need to be mutable? seems unfortunate to stick a mutable thing in place here